### DISCUSSION TITLE
Transition of animated wallpaper

### DISCUSSION THREAD

**Discussion by MiHaElVela** [Taiizor/Sucrose#85] [Ideas]
Hello, I appreciate the app, and I would like to suggest a new function, the transition between wallpapers.

**Comment by Taiizor** [comment]
If you mean switching between wallpapers in your library after a set period, I'm working on it, but it might take a while as I haven't had much time lately.

**Reply by MiHaElVela** [reply]
I trust your work. Thank you for the project. I'll wait patiently

**Reply by Taiizor** [reply]
If I need to provide a brief overview of this feature:

* Transition time between wallpapers (in minutes) can be adjusted.
* Transitions between wallpapers can be set to random or sequential.
* Each theme in your library can be included or excluded from automatic transitions.

If you have any other thoughts or ideas about this feature, please let me know.

**Comment by kevin122673** [comment]
Transition Effects
 - Cut
 -  Fade
 -  Dissolve
 -  Wipe

Wipe Transition Settings
 - Left
 - Right
 - Up
 - Down

Wait until end of theme before transitioning. [ OFF | ON ]
     This option does not apply to dynamic wallpapers
  
--

For the sequential could it follow the sort order?

--
If you don't want an added included/excluded overlay element, perhaps handle the include as a playlist. This would allow independent sorting in the playlist.
![sucrose-playlist](https://github.com/Taiizor/Sucrose/assets/103470356/e4b7b68d-006f-4f3f-95d3-1ffb24b63069)

**Reply by Taiizor** [reply]
Transition between wallpapers is not like a slideshow, so creating transition effects is not really feasible.

I didn't quite understand the "Wipe Transition Settings" part, to be honest.

The situation where the current theme has finished or not when the transition is triggered will only apply to Gif and Video themes. So, I don't want to pay much attention to this situation.

In random transition, the next theme is already randomly chosen, but in sequential transition, it will follow the order of the themes displayed in your library. You can also change how themes are sorted in your library from the setting shown in the image below:

![Sucrose](https://github.com/Taiizor/Sucrose/assets/41683699/3624a48c-a94c-4422-b9ff-2496b692db35)

Themes will be included in the transition by default, but if someone doesn't want this, they can exclude the relevant theme, similar to the menu option you showed in your image. More thought can be put into creating playlists, but I don't want this feature to become too complicated.

**Comment by kevin122673** [comment]
The use of cut transitions is a good starting point for the feature. 

A wipe transition has direction. See this video: https://youtu.be/wG4TMXKKUck?si=iEz0G2obuJQKzcpO&t=75
*The setting is moot until other transition effects are added.

**Comment by Taiizor** [comment]
The menu view will look like the following:

![Sucrose](https://github.com/Taiizor/Sucrose/assets/41683699/3a977e51-a4e4-4643-8f69-43da9e3f30f8)
![Sucrose](https://github.com/Taiizor/Sucrose/assets/41683699/dacbf54e-8397-48c0-a032-d7bfa55d8fb8)

**Reply by Taiizor** [reply]
The interface settings also appear as follows:

![Sucrose](https://github.com/Taiizor/Sucrose/assets/41683699/c0e51228-640e-42c8-ab66-faf9cd945776)

**Reply by Taiizor** [reply]
The wallpaper cycyling feature will be available in the next release.
